src/metabolic_ninja/tasks.py

```diff
@@ -117,13 +122,31 @@ def predict(self, model_obj, product_name, max_predictions, aerobic,
     #  default_biomass_reaction. Maybe we need a new field for the medium
     #  database model?
     model.carbon_source = "EX_glc__D_e"
+    # Retrieve user details. This should be retrieved here, to ensure that the
+    # JWT has not expired by the time it is used.
+    response = requests.get(f"{os.environ['IAM_API']}/user", headers={
+        'Authorization': f"Bearer {token}",
+    })
+    response.raise_for_status()
+    user = response.json()
+    user_name = f"{user['first_name']} {user['last_name']}"
+    user_email = user['email']
+    # Retrieve the organism name.
+    response = requests.get(
+        f"{os.environ['WAREHOUSE_API']}/organisms/{organism_id}",
+        headers={'Authorization': f"Bearer {token}"},
+    )
+    response.raise_for_status()
+    organism_name = response.json()['name']
     # Define the workflow.
     workflow = (
         find_product.s(product_name, source) |
         find_pathways.s(model, max_predictions, source) |
         optimize.s(model) |
         concatenate.s() |
-        persist.s(job_id)
+        persist.s(job_id) |
+        notify.s(job_id, product_name, organism_id, user_name, user_email,
```

Comment:
If you use the immutable signature instead, celery will not pass on the previous result. You can thus avoid accepting `result` on line 408 below and also don't need to return it again.

```suggestion
        notify.si(job_id, product_name, organism_id, user_name, user_email,
```